File: SMI-SDK/.NET SDK/Library/gis/trace/HTTPRequestMessageEntryProperty.cs
namespace pt.sapo.gis.trace
{
    using System;
    using System.Collections.Generic;

    /**
    * @author GIS Team 
     * Entry property to wrap an HTTP request message
     */
    public class HTTPRequestMessageEntryProperty : HTTPMessageEntryProperty
    {
        public static readonly String URL_PROPERTY_NAME = "url";
        public static readonly String VERB_PROPERTY_NAME = "verb";
        
        public String Verb
        {
            get { return (String)this[VERB_PROPERTY_NAME]; }
            set { this.Add(VERB_PROPERTY_NAME, value); }
        }

        public Uri Url
        {
            get { return (Uri)this[URL_PROPERTY_NAME]; }
            set { this.Add(URL_PROPERTY_NAME, value); }
        }
    }
}

File: SMI-SDK/Java SDK/src/pt/sapo/gis/trace/HTTPRequestMessageEntryProperty.java
package pt.sapo.gis.trace;

/**
* @author GIS Team 
 * Entry property to wrap an HTTP request message
 */
public class HTTPRequestMessageEntryProperty extends HTTPMessageEntryProperty{
    public static final String PATH_AND_QUERY_PROPERTY_NAME = "pathAndQuery";
    public static final String VERB_PROPERTY_NAME = "verb";
    public static final String SECURE_PROPERTY_NAME = "secure";
    
   public String getPathAndQuery() {
        return (String)this.get(PATH_AND_QUERY_PROPERTY_NAME);
    }

    public void setPathAndQuery(String pathAndQuery) {
        this.put(PATH_AND_QUERY_PROPERTY_NAME, pathAndQuery);
    }

    public boolean isSecure() {
        return ((Boolean)this.get(SECURE_PROPERTY_NAME)).booleanValue();
    }

    public void setSecure(boolean secure) {
        this.put(SECURE_PROPERTY_NAME, Boolean.valueOf(secure));
    }

    public String getVerb() {
        return (String)this.get(VERB_PROPERTY_NAME);
    }

    public void setVerb(String verb) {
        this.put(VERB_PROPERTY_NAME, verb);
    }
}